mm2src/mm2_main/src/lp_swap/taker_swap.rs

```diff
@@ -1829,9 +1845,40 @@ impl TakerSwap {
             tx_hash,
         };
 
-        Ok((Some(TakerSwapCommand::Finish), vec![TakerSwapEvent::MakerPaymentSpent(
-            tx_ident,
-        )]))
+        Ok((Some(TakerSwapCommand::ConfirmMakerPaymentSpend), vec![
+            TakerSwapEvent::MakerPaymentSpent(tx_ident),
+        ]))
+    }
+
+    async fn confirm_maker_payment_spend(&self) -> Result<(Option<TakerSwapCommand>, Vec<TakerSwapEvent>), String> {
+        // we should wait for only one confirmation to make sure our spend transaction is not failed
+        let confirm_maker_payment_spend_input = ConfirmPaymentInput {
+            payment_tx: self.r().maker_payment_spend.clone().unwrap().tx_hex.0,
+            confirmations: std::cmp::min(1, self.r().data.maker_payment_confirmations),
+            requires_nota: self.r().data.maker_payment_requires_nota.unwrap_or(false),
```

Comment:
@shamardy when we wait for taker payment spend on maker side, we use `false` for `requires_nota`.
Shouldnt we do the same on taker side, when they wait for maker payment spend?

https://github.com/KomodoPlatform/komodo-defi-framework/blob/b9ba150e04e328172098b5bc72c8d7182e62151e/mm2src/mm2_main/src/lp_swap/maker_swap.rs#L1143-L1153

Reply:
> when we wait for taker payment spend on maker side, we use false for requires_nota.
Shouldnt we do the same on taker side, when they wait for maker payment spend?

Agree, since we use 1 confirmation at most. I was wrong to suggest otherwise in a previous review.